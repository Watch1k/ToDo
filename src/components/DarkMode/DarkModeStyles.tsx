import React from 'react';

export const darkModeStyles: React.CSSProperties = {
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
}
