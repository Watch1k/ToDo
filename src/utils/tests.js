import React from 'react'
import { Provider } from 'react-redux'
import { render } from '@testing-library/react'

export const renderWithRedux = (ui, store) => ({
  ...render(<Provider store={store}>{ui}</Provider>),
  store,
})
