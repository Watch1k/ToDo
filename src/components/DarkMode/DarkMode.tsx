import React from 'react'
import { Switch } from 'antd'
import { useDarkModeConnect } from '../../useConnect'
import { darkModeStyles } from './DarkModeStyles'

const DarkMode = () => {
  const { darkMode, setDarkMode } = useDarkModeConnect()
  const handleChange = (checked: boolean) => {
    setDarkMode(checked)
  }

  return (
    <div style={darkModeStyles}>
      <Switch
        onChange={handleChange}
        checked={darkMode}
        data-testid='darkMode'
      />
    </div>
  )
}

export default DarkMode
