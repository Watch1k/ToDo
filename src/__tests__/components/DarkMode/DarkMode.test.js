import React from 'react';
import { fireEvent } from '@testing-library/react'
import { store } from '../../../store'
import { renderWithRedux } from '../../../utils/tests'
import DarkMode from '../../../components/DarkMode/DarkMode'

describe('DarkMode component', () => {
  const wrapper = renderWithRedux(<DarkMode />, store);

  it('Renders', () => {
    expect(wrapper.getByTestId('darkMode')).toBeInTheDocument();
  });

  it('Matches snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('DarkMode state should be true after click', () => {
    fireEvent.click(wrapper.getByTestId('darkMode'))
    expect(store.getState().darkMode).toBe(true)
  })
});
