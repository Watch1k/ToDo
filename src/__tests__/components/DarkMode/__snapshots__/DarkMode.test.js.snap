// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkMode component Matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-direction: column; align-items: center;"
  >
    <button
      aria-checked="false"
      class="ant-switch"
      data-testid="darkMode"
      role="switch"
      type="button"
    >
      <span
        class="ant-switch-inner"
      />
    </button>
  </div>
</DocumentFragment>
`;
